tests/foreman/api/test_audit.py

```diff
@@ -93,9 +92,13 @@ def test_positive_create_by_type(self):
                 'entity_type', created_entity.__class__.__name__.lower())
             value_template = entity_item.get('value_template', '{entity.name}')
             entity_value = value_template.format(entity=created_entity)
-            audit = entities.Audit().search(
-                query={'search': 'type={0}'.format(entity_type)})[0]
-            self.assertEqual(audit.auditable_name, entity_value)
+            audits = entities.Audit().search(
+                query={'search': 'type={0}'.format(entity_type)})
+            try:
+                audit = next(entry for entry in audits
+                             if entry.auditable_name == entity_value)
+            except StopIteration:
+                raise Exception('audit not found by name')
```

Comment:
perhaps like this will be more readable
```python
entity_audits = [entry for entry in audits if entry.auditable_name == entity_value]
if not entity_audits:
    raise Exception('audit not found by name "{}"'.format(entity_value))
audit = entity_audits[0]
```
we create a new var but think more readable, an other think if we cannot find the entity should we raise Exception or fail the test, perhaps calling self.fail will be the more appropriate approach, what do you think?